### PR TITLE
build: use external sortition package

### PR DESCRIPTION
**Pull request by cce**
## Summary

This updates the data/committee package to use the new separate [github.com/algorand/sortition](https://github.com/algorand/sortition) package, which contains the same code as in github.com/algorand/data/committee/sortition.

## Test Plan

Existing tests should pass, and added a new property-based test TestCompareSortitionImpls that I ran for 10M iterations.

### CODE REVIEW

**Review by algorandskiy — APPROVED** [review]

**Review by bbroder-algo — APPROVED** [review]
lgtm

**Review by tzaffi — COMMENTED** [review]
LGTM

**Review by tzaffi — DISMISSED** [review]

File: data/committee/sortition/sortition_test.go [hunk 1]
```diff
@@ -62,3 +68,57 @@ func TestSortitionBasic(t *testing.T) {
 		t.Errorf("wanted %d selections but got %d, d=%d, maxd=%d", expected, hitcount, d, maxd)
 	}
 }
+
+func TestCompareSortitionImpls(tt *testing.T) {
+	partitiontest.PartitionTest(tt)
+
+	// if TOTAL_MONEY env var is set, parse uint64 from env
+	var envTotalMoney uint64
+	if tm := os.Getenv("TOTAL_MONEY"); tm != "" {
```
Comment (tzaffi) [inline data/committee/sortition/sortition_test.go:77]: The env var is for configuring the unit test locally? Is it worthwhile ranging over more than one value in CI?

Comment (tzaffi) [reply data/committee/sortition/sortition_test.go:77]: So a non-blocking suggestions: consider removing the env-var logic as you can just hard code it as necessary locally.

**Comment by algorandskiy** [comment]
@cce please update go.sum

File: data/committee/sortition/sortition_test.go [hunk 2]
```diff
@@ -62,3 +68,57 @@ func TestSortitionBasic(t *testing.T) {
...
+		expectedSize := rapid.OneOf(
+			rapid.Just(proto.NumProposers),
+			rapid.Just(proto.SoftCommitteeSize),
+			rapid.Just(proto.CertCommitteeSize),
+			rapid.Just(proto.NextCommitteeSize),
+			rapid.Just(proto.LateCommitteeSize),
+			rapid.Just(proto.RedoCommitteeSize),
+			rapid.Just(proto.DownCommitteeSize),
+		).Draw(t, "expectedSize").(uint64)
+		//expectedSize := rapid.Uint64Range(1, totalMoney).Draw(t, "expectedSize").(uint64) // draw random
```
Comment (tzaffi) [inline data/committee/sortition/sortition_test.go:96]:
```suggestion
```

Or if you really need for local testing, it should be placed in the scope of `totalMoney`

**Review by tzaffi — COMMENTED** [review]
left a nit about removing a commented out line in a test

**Review by winder — APPROVED** [review]

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**